src/generic-container/started-generic-container.ts

```diff
@@ -35,6 +57,24 @@ export class StartedGenericContainer implements StartedTestContainer {
     return new StoppedGenericContainer();
   }
 
+  private async waitForContainer(container: Dockerode.Container, boundPorts: BoundPorts): Promise<void> {
```

Comment:
This is duplicated from `generic-container`, perhaps we should extract this to a dedicated `wait-for-container` module, which accepts the wait strategy

Reply:
Maybe we should open another PR for this instead? Since it make changes across multiple files, would like to keep the PRs focused. I can open a seperate one for it.